cmd/siad/server.go

```diff
@@ -422,9 +578,19 @@ func (srv *Server) Serve() error {
 
 // Close closes the Server's listener, causing the HTTP server to shut down.
 func (srv *Server) Close() error {
+	var errs []error
 	// Close the listener, which will cause Server.Serve() to return.
 	if err := srv.listener.Close(); err != nil {
-		return err
+		errs = append(errs, err)
 	}
-	return nil
+	// Close all of the modules
+	for i := len(srv.moduleClosers) - 1; i >= 0; i-- {
+		m := srv.moduleClosers[i]
+		fmt.Printf("Closing %v...\n", reflect.TypeOf(m))
```

Comment:
does this actually print the right thing?

Reply:
Example output:

```
Caught stop signal, quitting...
Closing *renter.Renter...
Closing *host.Host...
Closing *wallet.Wallet...
Closing *transactionpool.TransactionPool...
Closing *consensus.ConsensusSet...
Closing *gateway.Gateway...
Shutdown complete.
```

Reply:
It's a cute trick but not good enough. Let's use a separate slice for the names, or define a type containing both the name and io.Closer.

Reply:
ACK, I added a new `moduleCloser` type which prints the name in its `Close` method.